### PR TITLE
Bug 1821961 - Updated the collapse animation of the tabs tray FAB & scrim

### PR DESCRIPTION
**Pull request by t-p-white**
Demo

https://user-images.githubusercontent.com/1833156/233059667-c0e789aa-9cb5-4e55-bae7-d1a13cfc2f49.mov

Note: This non-Compose implementation of tabs tray will be phased out as Compose becomes the primary layout. 

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1821961

### CODE REVIEW

**Comment by t-p-white** [comment]
@Verdi could you kindly give a UX review please? Please see attached video demo.

**Comment by Verdi** [comment]
> @Verdi could you kindly give a UX review please? Please see attached video demo.

This looks great - thank you

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt [hunk 1]
```diff
@@ -312,8 +316,25 @@ class TabsTrayFragment : AppCompatDialogFragment() {
...
             tabsTrayBinding.tabWrapper
         }
+
+        val newTabFab = if (requireContext().settings().enableTabsTrayToCompose) {
+            fabButtonComposeBinding.root
+        } else {
+            fabButtonBinding.newTabButton
+        }
+
+        val behavior = BottomSheetBehavior.from(rootView)
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:326]: nit: The behavior could alternatively be instantiated like:
```suggestion
        val behavior = BottomSheetBehavior.from(rootView).apply {
            addBottomSheetCallback(traySheetBehaviorCallback)
            skipCollapsed = true
        }
```

File: fenix/app/src/test/java/org/mozilla/fenix/tabstray/TabSheetBehaviorManagerTest.kt [hunk 2]
```diff
@@ -67,35 +68,409 @@ class TabSheetBehaviorManagerTest {
...
+    @Test
+    fun `GIVEN behaviour state is 'dragging' & draggedLowestSheetTop is null WHEN onSlide is called THEN draggedLowestSheetTop is set to currentBottomSheetTop`() {
+        // Behavior
+        val behavior = mockk<BottomSheetBehavior<ConstraintLayout>>(relaxed = true)
+        every { behavior.state } returns STATE_DRAGGING
+
+        // Dialog
+        val tabsTrayDialog = mockk<TabsTrayDialog>(relaxed = true)
+
+        // Fab
```
Comment (MozillaNoah) [inline fenix/app/src/test/java/org/mozilla/fenix/tabstray/TabSheetBehaviorManagerTest.kt:154]: nit: The comments for the initialization blocks (behavior, dialog, fab) in these test aren't really necessary.

**Review by MozillaNoah — APPROVED** [review]
Left a few nits, but this looks good to go otherwise! You added a nice set of tests.

File: fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt (hunk 1 above)
Comment (t-p-white) [reply fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:326]: nice suggestion 👍

**Pull request merged**